calico.py

```diff
@@ -38,6 +38,9 @@
 from calico_cni.policy_drivers import ApplyProfileError, get_policy_driver
 from ipam import IpamPlugin
 
+# Hostname which can be overridden with config
+HOSTNAME = socket.gethostname()
```

Comment:
This value is only used within the `CniPlugin` class.  Why don't we define this within in the class (like the rest of the config options) so we don't need to do this global stuff.

This should reduce the number of lines, and make it easier to manipulate in UTs.

```
self.hostname = config.get("hostname", socket.gethostname())
```

Or something like that.